src/System.Private.CoreLib/shared/System/Globalization/ISOWeek.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using static System.DateTime;
+
+namespace System.Globalization
+{
+    public static class ISOWeek
+    {
+        private const int WeeksInLongYear = 53;
+        private const int WeeksInShortYear = 52;
+
+        private const int MinWeek = 1;
+        private const int MaxWeek = WeeksInLongYear;
+
+        public static int GetWeekOfYear(DateTime date)
+        {
+            int week = GetWeekNumber(date);
+
+            if (week < MinWeek)
+            {
+                return GetWeeksInYear(date.Year - 1);
+            }
+
+            if (week > GetWeeksInYear(date.Year))
+            {
+                return 1;
+            }
+
+            return week;
+        }
+
+        public static int GetYear(DateTime date)
+        {
+            int week = GetWeekNumber(date);
+
+            if (week < MinWeek)
+            {
+                return date.Year - 1;
+            }
+
+            if (week > GetWeeksInYear(date.Year))
+            {
+                return date.Year + 1;
+            }
+
+            return date.Year;
+        }
+
+        public static DateTime GetYearStart(int year)
```

Comment:
>GetYearStart [](start = 31, length = 12)

could you please add a comment to all methods taking a year telling it is the ISO Week year (and not necessary matching the Gregorian year number)?

Reply:
That only applies to `GetYearStart` and `GetYearEnd`, right?

Reply:
right